core/src/main/python/wlsdeploy/tool/deploy/applications_deployer.py

```diff
@@ -1079,6 +1129,9 @@ def __deploy_app_online(self, application_name, source_path, targets, stage_mode
                 kwargs[key] = value
         kwargs['timeout'] = self.model_context.get_model_config().get_deploy_timeout()
 
+        if module_type in ['jms', 'jdbc', 'wldf' ] and sub_module_targets is not None:
```

Comment:
Can the `module_type` check be replaced with a call to `self.version_helper.is_module_type_app_module(module_type)` ?
Just to be consistent with other checks for this.